integration/test/performancetest/transmitter.go

```diff
@@ -0,0 +1,239 @@
+	"github.com/aws/aws-sdk-go-v2/feature/dynamodb/attributevalue"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+)
+
+const METRIC_PERIOD = agentRuntimeMinutes * 60 // this const is in seconds , 5 mins
+
+type TransmitterAPI struct {
+	dynamoDbClient *dynamodb.Client
+	DataBaseName   string // this is the name of the table when test is run
+}
+
+// this is the packet that will be sent converted to DynamoItem
+type Metric struct {
+	Average float64
+	P99     float64 //99% percent process
+	Max     float64
+	Min     float64
+	Period  int //in seconds
+	Data    []float64
+}
+
+type collectorData []struct { // this is the struct data collector passes in
+	Id         string    `json:"Id"`
+	Label      string    `json:Label`
+	Messages   string    `json:Messages`
+	StatusCode string    `json:StatusCode`
+	Timestamps []string  `json:Timestamps`
+	Values     []float64 `json:Values`
+}
+
+/*
+InitializeTransmitterAPI
+Desc: Initializes the transmitter class
+Side effects: Creates a dynamodb table if it doesn't already exist
+*/
+func InitializeTransmitterAPI(DataBaseName string) *TransmitterAPI {
+	//setup aws session
+	cfg, err := config.LoadDefaultConfig(context.TODO(), config.WithRegion(os.Getenv("AWS_REGION")))
+	if err != nil {
+		fmt.Printf("Error: Loading in config %s\n", err)
+	}
+	transmitter := TransmitterAPI{
+		dynamoDbClient: dynamodb.NewFromConfig(cfg),
+		DataBaseName:   DataBaseName,
+	}
+	// check if the dynamo table exist if not create it
+	tableExist, err := transmitter.TableExist()
+	if err != nil {
+		return nil
+	}
+	if !tableExist {
+		fmt.Println("Table doesn't exist")
+		err := transmitter.CreateTable()
+		if err != nil {
+			return nil
+		}
+	}
+	fmt.Println("API ready")
+	return &transmitter
+
+}
+
+/*
+CreateTable()
+Desc: Will create a DynamoDB Table with given param. and config
+*/
+ //add secondary index space vs time  
+func (transmitter *TransmitterAPI) CreateTable() error {
+	_, err := transmitter.dynamoDbClient.CreateTable(
+		context.TODO(), &dynamodb.CreateTableInput{
+			AttributeDefinitions: []types.AttributeDefinition{
+				{
+					AttributeName: aws.String("Hash"),
+					AttributeType: types.ScalarAttributeTypeS,
+				},
+				{
+					AttributeName: aws.String("CommitDate"),
+					AttributeType: types.ScalarAttributeTypeN,
+				},
+			},
+			KeySchema: []types.KeySchemaElement{
+				{
+					AttributeName: aws.String("Hash"),
+					KeyType:       types.KeyTypeHash,
+				},
+				{
+					AttributeName: aws.String("CommitDate"),
+					KeyType:	   types.KeyTypeRange,
+				},
+			},
+			ProvisionedThroughput: &types.ProvisionedThroughput{
+				ReadCapacityUnits:  aws.Int64(10),
+				WriteCapacityUnits: aws.Int64(10),
+			},
+			TableName: aws.String(transmitter.DataBaseName),
+		}) // this is the config for the new table)
+	if err != nil {
+		fmt.Printf("Error calling CreateTable: %s", err)
+		return err
+	}
+	time.Sleep(10*time.Second)
```

Comment:
Try to see if there is a way to have wait for create instead of sleep maybe a polling for loop